service/src/main/java/bio/terra/cbas/initialization/InstanceInitializer.java

```diff
@@ -8,16 +9,25 @@
 
 @Component
 public class InstanceInitializer implements ApplicationListener<ContextRefreshedEvent> {
-
   private final BackfillOriginalWorkspaceIdService backfillOriginalWorkspaceIdService;
+  private final CloneRecoveryService cloneRecoveryService;
+
+  private final CbasInitializationConfiguration cbasInitializationConfiguration;
 
   public InstanceInitializer(
-      BackfillOriginalWorkspaceIdService backfillOriginalWorkspaceIdService) {
+      CbasInitializationConfiguration cbasInitializationConfiguration,
+      BackfillOriginalWorkspaceIdService backfillOriginalWorkspaceIdService,
+      CloneRecoveryService cloneRecoveryService) {
+    this.cbasInitializationConfiguration = cbasInitializationConfiguration;
     this.backfillOriginalWorkspaceIdService = backfillOriginalWorkspaceIdService;
+    this.cloneRecoveryService = cloneRecoveryService;
   }
 
   @Override
   public void onApplicationEvent(@NotNull ContextRefreshedEvent event) {
-    backfillOriginalWorkspaceIdService.backfillAll();
+    if (cbasInitializationConfiguration.getEnabled()) {
+      backfillOriginalWorkspaceIdService.backfillAll();
+      cloneRecoveryService.cloneRecovery();
```

Comment:
Is there any way to check that backfilling succeeded before rolling in to recovery? Or validation like "there are no null originalWorkspaceIds any more"?